src/rdf4cpp/rdf/datatypes/registry/LiteralDatatypeImpl.hpp

```diff
@@ -181,17 +182,41 @@ struct Comparable {
     }
 };
 
+/**
+ * The capability to have a fixed id and therefore
+ * be looked up in constant time
+ */
+template<util::ConstexprString type_iri>
+struct FixedId {
+    static constexpr LiteralType fixed_id = reserved_datatype_ids[type_iri];
+    static_assert(fixed_id.is_fixed(), "cannot treat non fixed id as fixed");
+};
+
 } // namespace capabilities
 
 /**
  * An automatically registering LiteralDatatype with given capabilities.
  *
  * @tparam Capabilities all capabilities this instantiation should have
  */
-template<ConstexprString type_iri, template<ConstexprString> typename... Capabilities>
+template<util::ConstexprString type_iri, template<util::ConstexprString> typename... Capabilities>
 struct LiteralDatatypeImpl : capabilities::Default<type_iri>, Capabilities<type_iri>... {
     using typename capabilities::Default<type_iri>::cpp_type;
 
+    // todo: find better name
+    static constexpr DatatypeIRIView const datatype_iri = []() {
```

Comment:
LiteraldatatypeImpl::datatype_iri probably also needs a better name

Reply:
I would probably make that dependent on the name of the type